### DISCUSSION TITLE
How to create a field to expose Model Fields' choices

### DISCUSSION THREAD

**Discussion by ZuluPro** [graphql-python/graphene-django#1180] [Q&A]
I have a model field with choices: 

```
CHOICES = [
  ('foo', 'Foo'),
  ('bar', 'Bar'),
]

class MyModel(models.Model):
    ham = models.CharField(max_length=2, choices=CHOICES)
```

So, in the schema I have the model and its choices. But can I expose the choices ?
I tried this but without success:
```
MyModelHamType = registry.get_converted_field(models.MyModel._meta.get_field('ham')).get_type()

class Query(object):
    choices = graphene.Field(MyModelHamType)

    def resolve_choices(self, info, **kwargs):
        # What to do here
        return MyModelHamType
```

The code above gives the following thing: 
Request : 
```
query {
  hams
}
```

Response: 
```
{
  "errors": [
    {
      "message": "Expected a value of type \"MyModelHam\" but received: MyModelHam",
      "path": [
        "hams"
      ]
    }
  ],
  "data": {
    "hams": null
  }
}
```

It seems that as the object aren't the of the same type (Pythonicly saying), Graphene doesn't want to handle.

----

I could recreate myself a `MyModelHamType` with `Enum`, but it will be duplicate with the existing one autocreated and already in schema.

**Comment by ZuluPro** [comment]
@sciyoshi Any news ?

**Comment by dopeboy** [comment]
Let's take this over to #67. Closing this down.

**Comment by OlafHaalstra** [comment]
@ZuluPro were you able to fix this issue? Could you explain how?
